caffe2/operators/utility_ops.cu

```diff
@@ -36,12 +36,27 @@ bool SumOp<CUDAContext>::RunOnDevice() {
   return false;
 }
 
+template <>
+bool SumGradientOp<CUDAContext>::RunOnDevice() {
+  CAFFE_ENFORCE_GT(InputSize(), 1);
```

Comment:
Since this is inside SumGradientOp instead of SumOp, the first input corresponds to dY (Y being the output of SumOp) and second input corresponds to first input of SumOp, right?

Reply:
@zqhsfz2 could you explain in your SumGradient, what are input(s) and output(s)?

Reply:
Sure. Assuming what SumOp does is Y = X1 + X2 + X3 + ... Xn, the inputs for SumGradientOp would be (dY, X1, X2, ..., Xn) and outputs for SumGradientOp would be (dX1, dX2, ..., dXn). This has been specified in L19-25 of:

https://github.com/pytorch/pytorch/pull/16485/files#diff-1e7bf4989a4beddf9c0c537e29f47ef0R16